plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizardModel.java

```diff
@@ -104,14 +103,16 @@ private void updateSelectedAppSource(boolean useLocalAppSource, IApplicationSour
 		firePropertyChange(PROPERTY_SELECTED_APP_SOURCE, oldSelectedAppSource, this.selectedAppSource);
 	}
 
-	private void updateLabels(IApplicationSource source) {
-		if (source != null && ResourceKind.TEMPLATE.equals(source.getKind())) {
-			ITemplate template = (ITemplate) source.getSource();
-			setLabels(template.getObjectLabels());
-			return;
-		}
-		setLabels(Collections.emptyMap());
-	}
+    private void updateLabels(IApplicationSource source) {
+        if (source != null) {
+            if (ResourceKind.TEMPLATE.equals(source.getKind())) {
+                ITemplate template = (ITemplate)source.getSource();
+                setLabels(template.getObjectLabels());
+            } else {
+                setLabels(Arrays.asList(new Label("app", source.getSource().getName())));
```

Comment:
can we please have a speaking constant for "app"? It would expose this hard-coded label in a prominent place right on top of the class (and a speaking name would unveil what it is used for)